Add v3.3 documents #939 (#943)
slackapi/python-slack-sdk: docs-src/audit-logs/index.rst

==============================================
Audit Logs API Client
==============================================

`Audit Logs API <https://api.slack.com/admins/audit-logs>`_ is a set of APIs for monitoring what’s happening in your `Enterprise Grid <https://api.slack.com/enterprise/grid>`_ organization.

The Audit Logs API can be used by security information and event management (SIEM) tools to provide an analysis of how your Slack organization is being accessed. You can also use this API to write your own applications to see how members of your organization are using Slack.

> Follow the instructions in `the API document <https://api.slack.com/admins/audit-logs>`_ to get a valid token for using Audit Logs API. Your Slack app for Audit Logs API needs to be installed on the Enterprise Grid Organization, not an individual workspace within the organization.

Reviewer: ```suggestion
Follow the instructions in `the API document <https://api.slack.com/admins/audit-logs>`_ to get a valid token for using Audit Logs API. The Slack app using the Audit Logs API needs to be installed in the Enterprise Grid Organization, not an individual workspace within the organization.
```

AuditLogsClient
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

An OAuth token with `the admin scope <https://api.slack.com/scopes/admin>`_ is required to access this API.

> Probably, you are going to use the ``/logs`` endpoint. To learn available parameters for the endpoint, check `the guide <https://api.slack.com/admins/audit-logs#how_to_call_the_audit_logs_api>`_. Also, you can check the data structure of ``api_response.typed_body`` `here <https://github.com/slackapi/python-slack-sdk/blob/main/slack_sdk/audit_logs/v1/logs.py>`_.

Reviewer: ```suggestion
You will likely use the ``/logs`` endpoint to ???. 

To learn about the available parameters for this endpoint, check out `this guide <https://api.slack.com/admins/audit-logs#how_to_call_the_audit_logs_api>`_. You can also learn more about the data structure of ``api_response.typed_body`` `here <https://github.com/slackapi/python-slack-sdk/blob/main/slack_sdk/audit_logs/v1/logs.py>`_.
```

The ??? should be replaced with a concise why, IMO. It's not clear to me as a reader why I'd be using `/logs` and this is a good place to underscore its purpose.

Author: Thanks. I don't think we need to mention detailed reasons here. Audit Logs API users should already know what it is. Also, I would like to avoid repeating duplicated topics with the API document.

.. code-block:: python

    import os
    from slack_sdk.audit_logs import AuditLogsClient

    client = AuditLogsClient(token=os.environ["SLACK_ORG_ADMIN_USER_TOKEN"])

    api_response = self.client.logs(action="user_login", limit=1)
    api_response.typed_body  # slack_sdk.audit_logs.v1.LogsResponse

> If you would like to access ``/schemes`` or ``/actions``, You can use the following methods.

Reviewer: ```suggestion
If you would like to access ``/schemes`` or ``/actions``, you can use the following methods:
```

==============================================
Audit Logs API Client
==============================================

`Audit Logs API <https://api.slack.com/admins/audit-logs>`_ is a set of APIs for monitoring what’s happening in your `Enterprise Grid <https://api.slack.com/enterprise/grid>`_ organization.

The Audit Logs API can be used by security information and event management (SIEM) tools to provide an analysis of how your Slack organization is being accessed. You can also use this API to write your own applications to see how members of your organization are using Slack.

Follow the instructions in `the API document <https://api.slack.com/admins/audit-logs>`_ to get a valid token for using Audit Logs API. The Slack app using the Audit Logs API needs to be installed in the Enterprise Grid Organization, not an individual workspace within the organization.

AuditLogsClient
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

An OAuth token with `the admin scope <https://api.slack.com/scopes/admin>`_ is required to access this API.

You will likely use the ``/logs`` endpoint as it's the essential part of this API.

> To learn about the available parameters for this endpoint, check out `this guide <https://api.slack.com/admins/audit-logs#how_to_call_the_audit_logs_api>`_. You can also learn more about the data structure of ``api_response.typed_body`` `here <https://github.com/slackapi/python-slack-sdk/blob/main/slack_sdk/audit_logs/v1/logs.py>`_.

Reviewer 2: Nit: "here" can be a mysterious link, so I'd suggest saying "this guide" or "on GitHub" or "from logs.py" to let the reader know where the link will go. I went with the same wording that was used in the SCIM doc:

```suggestion
To learn about the available parameters for this endpoint, check out `this guide <https://api.slack.com/admins/audit-logs#how_to_call_the_audit_logs_api>`_. You can also learn more about the data structure of ``api_response.typed_body`` from `the class source code <https://github.com/slackapi/python-slack-sdk/blob/main/slack_sdk/audit_logs/v1/logs.py>`_.
```

Author: @mwbrooks Thanks, great suggestion!
